packages/flutter_tools/lib/src/macos/application_package.dart

```diff
@@ -156,23 +156,29 @@ class BuildableMacOSApp extends MacOSApp {
   String get name => 'macOS';
 
   @override
-  String? applicationBundle(BuildMode buildMode) {
+  String? applicationBundle(BuildInfo buildInfo) {
     final File appBundleNameFile = project.nameFile;
     if (!appBundleNameFile.existsSync()) {
       globals.printError('Unable to find app name. ${appBundleNameFile.path} does not exist');
       return null;
     }
+
+    String bundleDirectory = sentenceCase(buildInfo.mode.name);
+    if(buildInfo.flavor != null) {
+      bundleDirectory += ' ${sentenceCase(buildInfo.flavor!)}';
```

Comment:
Avoid the bang by promoting a local.
```suggestion
    final String? flavor = buildInfo.flavor;
    if (flavor != null) {
      bundleDirectory += ' ${sentenceCase(flavor)}';
```

Wow I'd never seen this:
https://github.com/flutter/flutter/blob/997d43618dc73b2d01a86adaccf9dc9672953c03/packages/flutter_tools/lib/src/xcode_project.dart#L590-L594
I was going to say we shouldn't be reverse engineering these paths, we should let Xcode tell us to where it's building the app via build settings like `TARGET_BUILD_DIR`.  However the logic that's already here before this PR is doing something similar...